accounts/accounts_test.go

```diff
@@ -81,6 +81,31 @@ func TestSign(t *testing.T) {
 	}
 }
 
+func TestSignHash(t *testing.T) {
+	dir, am := tmpManager(t, true)
+	defer os.RemoveAll(dir)
+
+	pass := "passwd"
+	acc, err := am.NewAccount(pass)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if _, unlocked := am.unlocked[acc.Address]; unlocked {
+		t.Fatal("expected account to be locked")
+	}
+
+	_, err = am.SignWithPassphrase(acc.Address, pass, testSigData)
+	if err != nil {
```

Comment:
Nitpick, but please merge this if condition check with the above assignment. Cleaner and more concise code.

```
if _, err = am.SignWithPassphrase(acc.Address, pass, testSigData); err != nil {
    ...
}
```

Also please to the same for the other check a few lines below this.